### DISCUSSION TITLE
Can eBPF tamper with function parameters or return values ​​while the Golang process is running?

### DISCUSSION THREAD

**Discussion by KingsonKai** [cilium/ebpf#1137] [Q&A]
I am engaged in chaos engineering. I want to inject some exceptions into a golang process to mutate the parameters or return values ​​of the target function. Can it be implemented based on eBPF? Of course, it is also acceptable to restart the process if necessary. But it is best not to need to recompile

**Comment by lmb** [comment]
In theory yes, using uprobes. Changing return parameters via uretprobe probably won't work however: https://github.com/golang/go/issues/22008

Please discuss generic eBPF questions (not related to the library) on the Slack or Stack Overflow.
